dojo/models.py

```diff
@@ -1086,46 +1084,73 @@ class Endpoint(models.Model):
     tags = TagField(blank=True, force_lowercase=True, help_text="Add tags that help describe this endpoint. Choose from the list or add new tags. Press Enter key to add.")
 
     class Meta:
-        ordering = ['product', 'protocol', 'host', 'path', 'query', 'fragment']
+        ordering = ['product', 'host', 'protocol', 'port', 'userinfo', 'path', 'query', 'fragment']
         indexes = [
             models.Index(fields=['product', 'mitigated']),
         ]
 
+    def __init__(self, *args, **kwargs):
+        kwargs = self.verify_and_clean(**kwargs)
```

Comment:
when is __init__ executed, only for new model instances? Or also when an instance is loaded from the DB?

Reply:
It is used when an instance is imported (`python manage.py loaddata ...`), it is created (`e = Endpoint()`) but I don't know how does it work in case of loading from DB.

Reply:
But DD didn't fail if DB store a "broken" endpoint. So it looks, `verify_and_clean` is not called during loading from DB

Reply:
not sure about this one. pretty sure some people will run into problems loading fixtures with broken endpoints, but maybe it's OK to force them to fix those in the fixture?